### DISCUSSION TITLE
Spark on GCP Dataproc - AnalysisException: Can not load class uk.gov.moj.dash.linkage.JaroWinklerSimilarity, please make sure it is on the classpath

### DISCUSSION THREAD

**Discussion by mshearer0** [moj-analytical-services/splink#888] [General]
Does anyone have experience of using Splink on PySpark on a GCP Dataproc cluster?

I'm struggling to load the Jarowinkler jar file. I've tried adding to SPARK_DIST_CLASSPATH and as an init action as part of /etc/spark/conf.dist/spark-defaults.conf but no joy.

The error is:

```AnalysisException: Can not load class uk.gov.moj.dash.linkage.JaroWinklerSimilarity, please make sure it is on the classpath```

**Comment by mshearer0** [comment]
According to https://stackoverflow.com/questions/55739463/cant-add-jars-pyspark-in-jupyter-of-google-dataproc there isn't a built-in way to do this dynamically without effectively just editing spark-defaults.conf and restarting the kernel.

I copied the scala-udf-similarity-0.0.9.jar from ./splink/splink/files/spark_jars/ into the cloud storage bucket created with the cluster and then added:

```spark.jars=gs://<storage bucket>/scala-udf-similarity-0.0.9.jar```

into /etc/spark/conf.dist/spark-defaults.conf

That worked when i restarted the kernel but i'm sure there must be a better way.

**Reply by RobinL** [reply]
Thanks for following up on this.  Sorry we couldn't help, we haven't used Splink on GCP
